core/src/main/java/org/elasticsearch/index/query/QueryStringQueryBuilder.java

```diff
@@ -839,6 +867,28 @@ protected int doHashCode() {
                 timeZone == null ? 0 : timeZone.getID(), escape, maxDeterminizedStates);
     }
 
+    private Map<String, Float> allQueryableDefaultFields(QueryShardContext context) {
+        Collection<String> allFields = context.simpleMatchToIndexNames("*");
+        Map<String, Float> fields = new HashMap<>();
+        for (String fieldName : allFields) {
+            if (fieldName.startsWith("_")) {
```

Comment:
Maybe better to use MapperService.isMetadataField since "_" is a valid prefix for a user field name.